apps/glusterfs/volume_entry_allocate.go

```diff
@@ -63,6 +66,209 @@ func (v *VolumeEntry) allocBricksInCluster(db *bolt.DB,
 	}
 }
 
+func (v *VolumeEntry) getEntryfromBrickName(tx *bolt.Tx, brickname string) (brickEntry *BrickEntry, e error) {
+	var brickEntries []*BrickEntry
+	brickids := v.BricksIds()
+	for _, brickid := range brickids {
+		brick, err := NewBrickEntryFromId(tx, brickid)
+		if err != nil {
+			return nil, err
+		}
+		brickEntries = append(brickEntries, brick)
+	}
+
+	for _, brickentry := range brickEntries {
+		nodeEntry, err := NewNodeEntryFromId(tx, brickentry.Info.NodeId)
+		if err != nil {
+			return nil, err
+		}
+		if brickname == fmt.Sprintf("%v:%v", nodeEntry.Info.Hostnames.Storage[0], brickentry.Info.Path) {
+			return brickentry, nil
+		}
+	}
+	return nil, ErrNotFound
+}
+
+func (v *VolumeEntry) replaceBrickInVolume(db *bolt.DB, executor executors.Executor,
```

Comment:
This function is too large.  Please reorganize into one or two more functions.  Feel free to create a new xxx_replace.go file to place all these functions in.